File: 2128-ReversePrefixOfWord/2128-ReversePrefixOfWord.go
// Last updated: 07/11/2025, 13:41:06
func reversePrefix(word string, ch byte) string {
    j := strings.IndexByte(word, ch)
	if j < 0 {
		return word
	}
	s := []byte(word)
	for i := 0; i < j; i++ {
		s[i], s[j] = s[j], s[i]
		j--
	}
	return string(s)
    
}

File: 2128-ReversePrefixOfWord/2128-ReversePrefixOfWord.py
# Last updated: 07/11/2025, 13:41:03
class Solution:
    def reversePrefix(self, word: str, ch: str) -> str:
        # i = word.find(ch)
        # return word if i == -1 else word[i::-1] + word[i + 1 :]
        i = word.find(ch) + 1
        return word[:i][::-1] + word[i:]
        